watershed_workflow/condition.py

```diff
@@ -337,7 +338,7 @@ def __init__(self, cell, edges):
     assert (len(waterway.edges) == m2.num_edges)
 
     # delete the centroid info to force recalculation
-    m2.clear_geometry_cache()
+    # m2.clear_geometry_cache()
```

Comment:
This has to be done -- if you don't do this on moving points, the geometry is wrong.

Reply:
It is happening at the end of the function code. Isn't it to force the recalculation of the centroids outside of function?
It breaks frequently, I think where there are no centroids to clear? Should we provide the condition before "del self._centroids" in  _mesh.py L301_ so check if centroids are there?